### ISSUE TITLE
[BUG] Crash in [FIRLoggerWrapper logWithLevel:withService:withCode:withMessage:withArgs:] 

### ISSUE THREAD

**Issue by akshetpandey** [firebase/firebase-ios-sdk#502]
### [READ] Step 1: Are you in the right place?
Yes

### [REQUIRED] Step 2: Describe your environment

  * Xcode version: **Version 9.1 (9B55)**
  * Firebase SDK version: **Firebase/Core (4.5.0)**
  * Firebase Component: **Performance/FirebaseInstanceID**
  * Component version: **Firebase/Performance (4.5.0), FirebaseInstanceID (2.0.5)**

### [REQUIRED] Step 3: Describe the problem
Crash in instrumentation
**Stack trace:**

```
Exception name: EXC_BAD_ACCESS / KERN_INVALID_ADDRESS
1 libobjc.A.dylib objc_retain + 147872
2 thriller +[FIRLoggerWrapper logWithLevel:withService:withCode:withMessage:withArgs:] + 1181296
3 CoreFoundation __invoking___ + 1349228
4 CoreFoundation -[NSInvocation invoke] + 165576
5 thriller +[FIRInstanceIDLogger logWithLevel:withService:withCode:withMessage:withArgs:] + 1881096
6 thriller -[FIRInstanceIDLogger logFuncDebug:messageCode:msg:] + 1881532
7 thriller __51-[FIRInstanceIDAuthService performScheduledCheckin]_block_invoke + 1852564
8 thriller -[FIRInstanceIDAuthService notifyCheckinHandlersWithCheckin:error:] + 1854508
9 thriller __56-[FIRInstanceIDAuthService fetchCheckinInfoWithHandler:]_block_invoke + 1853892
10 thriller __69-[FIRInstanceIDCheckinService checkinWithExistingCheckin:completion:]_block_invoke + 1864944
11 thriller __InstrumentDataTaskWithRequestCompletionHandler_block_invoke_2 + 2138240
12 CFNetwork __75-[__NSURLSessionLocal taskForClass:request:uploadFile:bodyData:completion:]_block_invoke + 40848
13 CFNetwork __49-[__NSCFLocalSessionTask _task_onqueue_didFinish]_block_invoke + 136112
14 Foundation __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 951052
15 Foundation -[NSBlockOperation main] + 166368
16 Foundation -[__NSOperationInternal _start:] + 99868
17 libdispatch.dylib _dispatch_client_callout + 4164
18 libdispatch.dylib _dispatch_block_invoke_direct$VARIANT$mp + 34500
19 libdispatch.dylib _dispatch_client_callout + 4164
20 libdispatch.dylib _dispatch_block_invoke_direct$VARIANT$mp + 34500
21 libdispatch.dylib dispatch_block_perform$VARIANT$mp + 34160
22 Foundation __NSOQSchedule_f + 958436
23 libdispatch.dylib _dispatch_client_callout + 4164
24 libdispatch.dylib _dispatch_continuation_pop$VARIANT$mp + 37840
25 libdispatch.dylib _dispatch_async_redirect_invoke$VARIANT$mp + 31952
26 libdispatch.dylib _dispatch_root_queue_drain + 57796
27 libdispatch.dylib _dispatch_worker_thread3 + 57100
28 libsystem_pthread.dylib _pthread_wqthread + 4380
```

#### Steps to reproduce:
Crash happens during startup, is kind of rare.
Don't have any other repo steps atm.

#### Relevant Code:
N/A

**Comment by paulb777** [comment]
Perhaps a duplicate of #402

**Comment by akshetpandey** [comment]
Looks like it is, if thats the case, it should already be fixed in 4.6.0, let me release an version with that and see if I am still seeing it in wild...

**Comment by jegnux** [comment]
@akshetpandey as I said [here](https://github.com/firebase/firebase-ios-sdk/issues/402#issuecomment-345193777), v4.6.0 didn't fix the issue for me.

**Comment by akshetpandey** [comment]
Yes, but that other ticket is still open and my crash report doesn't add anything new yet. I am keeping an eye on the other ticket and also releasing an app update with 4.6.0 today so I will know more soon.

**Comment by khopwood** [comment]
fyi, We still see this crash with 4.7

**Comment by pmiotto** [comment]
FYI, same issue on 4.7. Lots of crashes on iOS 11

**Comment by pmiotto** [comment]
Crashed: com.apple.main-thread
0  libobjc.A.dylib                0x184dfc43c objc_msgSend + 28
1  AusOpen                        0x102cbaee4 +[FIRLoggerWrapper logWithLevel:withService:withCode:withMessage:withArgs:] + 4308250340
2  CoreFoundation                 0x185ba9cd0 __invoking___ + 144
3  CoreFoundation                 0x185a8856c -[NSInvocation invoke] + 292
4  AusOpen                        0x102cdb048 +[FIRInstanceIDLogger logWithLevel:withService:withCode:withMessage:withArgs:] + 4308381768
5  AusOpen                        0x102cdb1fc -[FIRInstanceIDLogger logFuncDebug:messageCode:msg:] + 4308382204
6  AusOpen                        0x102cdef98 -[FIRInstanceIDTokenManager fetchNewTokenWithAuthorizedEntity:scope:keyPair:options:handler:] + 4308397976
7  AusOpen                        0x102cce81c __65-[FIRInstanceID tokenWithAuthorizedEntity:scope:options:handler:]_block_invoke_2.94 + 4308330524
8  libdispatch.dylib              0x18551ea54 _dispatch_call_block_and_release + 24
9  libdispatch.dylib              0x18551ea14 _dispatch_client_callout + 16
10 libdispatch.dylib              0x18552b698 _dispatch_main_queue_callback_4CF$VARIANT$mp + 1016
11 CoreFoundation                 0x185b4a544 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12
12 CoreFoundation                 0x185b48120 __CFRunLoopRun + 2012
13 CoreFoundation                 0x185a67e58 CFRunLoopRunSpecific + 436
14 GraphicsServices               0x187914f84 GSEventRunModal + 100
15 UIKit                          0x18f1bc67c UIApplicationMain + 236
16 AusOpen                        0x1029f8c34 main (main.m:14)
17 libdyld.dylib                  0x18558456c start + 4

**Comment by paulb777** [comment]
@pmiotto The fix is in FirebaseInstanceID 2.0.8 which is the default for Firebase 4.8.0. Please update.
